paasta_tools/deployd/common.py

```diff
@@ -71,19 +71,25 @@ def exponential_back_off(failures, factor, base, max_time):
     return seconds if seconds < max_time else max_time
 
 
-def get_service_instances_with_changed_id(marathon_client, instances, cluster):
-    marathon_app_ids = list_all_marathon_app_ids(marathon_client)
+def get_service_instances_needing_update(marathon_client, instances, cluster):
+    marathon_apps = {app.id: app for app in get_all_marathon_apps(marathon_client)}
+    marathon_app_ids = marathon_apps.keys()
     service_instances = []
     for service, instance in instances:
         config = load_marathon_service_config_no_cache(service=service,
                                                        instance=instance,
                                                        cluster=cluster,
                                                        soa_dir=DEFAULT_SOA_DIR)
         try:
-            config_app_id = config.format_marathon_app_dict()['id']
+            config_app = config.format_marathon_app_dict()
+            app_id = '/{}'.format(config_app['id'])
```

Comment:
Yeah it was just being stripped in another function before the comparison. Now I need the marathon app objects too, not just their IDs.